docs/DeveloperGuide.md

```diff
@@ -1,38 +1,527 @@
 # Developer Guide
```

Comment:
Overall, I like the structure and the lay out of the DG. However, I think it could be better to standardizing the diagrams.